### PR TITLE
Parsing Libraries properly for LBA1 and fixed floor samples

### PR DESCRIPTION
**Pull request by xesf**
**Preview here:** https://pr-460.lba2remake.net

### CODE REVIEW

File: src/game/loop/animAction.ts [hunk 1]
```diff
@@ -96,9 +98,13 @@ export const LEFT_STEP = (_action, { game, scene, animState }) => {
 };
 
 export const RIGHT_STEP = (_action, { game, scene, animState }) => {
-    const floorSound = animState.floorSound;
+    const isLBA1 = getParams().game === 'lba1';
+    let floorSound = animState.floorSound;
+    if (animState.floorSound2) {
```
Comment (slwilliams) [inline src/game/loop/animAction.ts:103]: hmm I'm a bit confused what floorSound2 represents. It is correct that only the right step should use floorSound2?

File: src/resources/parsers/libraries.ts [hunk 2]
```diff
@@ -26,11 +26,14 @@ const parseLayout = (dataView: DataView, index: number) => {
...
         const type = dataView.getUint8(offset + (i * 4) + 1);
+        const shape = dataView.getUint8(offset + (i * 4));
+        const brick = dataView.getUint16(offset + (i * 4) + 2, true);
         blocks.push({
-            shape: dataView.getUint8(offset + (i * 4)),
+            shape,
             sound: bits(type, 0, 4),
-            groundType: bits(type, 4, 4),
-            brick: dataView.getUint16(offset + (i * 4) + 2, true)
+            sound2: isLBA1 ? bits(type, 4, 4) : null,
```
Comment (slwilliams) [inline src/resources/parsers/libraries.ts:34]: Could you add a comment here explaining what sound2 is?

File: src/resources/parsers/libraries.ts [hunk 3]
```diff
@@ -26,11 +26,14 @@ const parseLayout = (dataView: DataView, index: number) => {
...
+            groundType: isLBA1 ? (type & 0xFF) === 0xF1 ? 1 : 0 : bits(type, 4, 4),
```
Comment (slwilliams) [inline src/resources/parsers/libraries.ts:35]: unrelated, but in LBA1 there's only 2 groundTypes? 1 or 0?

**Review by slwilliams — APPROVED** [review]

File: src/game/loop/animAction.ts (hunk 1 above)
Comment (xesf) [reply src/game/loop/animAction.ts:103]: yeah, the idea is left uses sound1 and right uses sound2

File: src/resources/parsers/libraries.ts (hunk 3 above)
Comment (xesf) [reply src/resources/parsers/libraries.ts:35]: yes, only water exist in LBA1.

File: src/resources/parsers/libraries.ts (hunk 2 above)
Comment (xesf) [reply src/resources/parsers/libraries.ts:37]: Sure, just pushed the comments.

Perhaps we should start creating documentations for each of these file formats and link them to the parser.

**Pull request merged**